### PR TITLE
MODQM-77 - Support stub endpoint duplicate a MARC bibliographic record

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
As a part of duplicate a MARC bibliographic record feature this merge request(https://issues.folio.org/browse/MODQM-77) aimed to create a stub endpoint to POST already existing MARC record to be duplicated via "source-record-manager" module. The further implementation is planned in https://issues.folio.org/browse/MODQM-80

## Approach
* introduced new endpoint POST /records-editor/records in ModuleDescriptor, raml and yaml files
* added stub implementation with Response status 501 - NOT_IMPLEMENTED
* added test
* updated README file

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/qm/client/SRMChangeManagerClient.java [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-
 
 @FeignClient(value = "change-manager")
 public interface SRMChangeManagerClient {
 
   @GetMapping(value = "/parsedRecords", produces = MediaType.APPLICATION_JSON_VALUE)
   ParsedRecordDto getParsedRecordByInstanceId(@RequestParam("instanceId") String instanceId);
 
+  @PostMapping(value = "/parsedRecords")
+  void postParsedRecord(@RequestBody  ParsedRecordDto recordDto);
```
Comment (psmagin) [inline src/main/java/org/folio/qm/client/SRMChangeManagerClient.java:21]: This method isn't used anywhere. Should it be deleted in this case?

File: src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import org.folio.qm.domain.dto.QuickMarc;
-import org.folio.qm.rest.resource.RecordsApi;
-import org.folio.qm.service.MarcRecordsService;
+import javax.validation.Valid;
+import java.util.UUID;
```
Comment (psmagin) [inline src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java:12]: Imports are in incorrect order in all classes. The order should be:
- import static *
- import static org.folio.*
- import java/javax.*
- import  *
- import org.folio.*

File: src/main/resources/swagger.api/records-editor.yaml [hunk 3]
```diff
@@ -59,6 +59,26 @@ paths:
             type: string
             pattern: "[a-zA-Z]{2}"
             default: "en"
+    post:
+      responses:
+        '202':
+          description: "Request is accepted"
```
Comment (psmagin) [inline src/main/resources/swagger.api/records-editor.yaml:65]: I think that this endpoint should return `status` entity contains`qmRecordId`. I will fix it in my story if no objections.

File: src/main/java/org/folio/qm/client/SRMChangeManagerClient.java (hunk 1 above)
Comment (natali-zaitseva) [reply src/main/java/org/folio/qm/client/SRMChangeManagerClient.java]: removed

File: src/main/resources/swagger.api/records-editor.yaml (hunk 3 above)
Comment (natali-zaitseva) [reply src/main/resources/swagger.api/records-editor.yaml:65]: no objections, thanks

File: README.md [hunk 4]
```diff
@@ -14,6 +14,7 @@ quickMARC API provides the following URLs:
 |  Method | URL| Permissions  | Description  | 
 |---|---|---|---|
 | GET | /records-editor/records?instanceId={instanceId}  |records-editor.records.item.get   | Retrieves QuickMarcJson by instance's id  |
+| POST| /records-editor/records|records-editor.records.item.post| Duplicate existing MARK and Instance records|
```
Comment (dmtkachenko) [inline README.md:17]: "Duplicate" is not the right word here. The endpoint creates new Marc bib record and other related records

File: ramls/records-editor.raml [hunk 5]
```diff
@@ -32,38 +32,57 @@ traits:
...
+      500:
+        description: "Internal server error, e.g. due to misconfiguration"
+        body:
+          application/json:
+            type: error
+            example:
+              strict: false
+              value: !include examples/error.sample
+  post:
+    description: "Create a new MARC record based on existing record."
```
Comment (dmtkachenko) [inline ramls/records-editor.raml:68]:
```suggestion
    description: "Create a new MARC record based."
```

File: src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java [hunk 6]
```diff
@@ -26,6 +29,11 @@ public RecordsEditorRecordsApi(MarcRecordsService marcRecordsService) {
     return ResponseEntity.ok(quickMarc);
   }
 
+  @Override
+  public ResponseEntity<Void> recordsPost(@Valid QuickMarc quickMarc) {
```
Comment (dmtkachenko) [inline src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java:33]: Is this method supposed to return Void but not the newly created data?

File: src/main/resources/swagger.api/records-editor.yaml (hunk 3 above)
Comment (dmtkachenko) [reply src/main/resources/swagger.api/records-editor.yaml:65]: well, not sure that returning status is appropriate in this case. usually POST returns the same entity that has been created or it's id

File: README.md (hunk 4 above)
Comment (natali-zaitseva) [reply README.md]: substituted to a "Create"

File: ramls/records-editor.raml (hunk 5 above)
Comment (natali-zaitseva) [reply ramls/records-editor.raml]: updated

File: src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java (hunk 6 above)
Comment (natali-zaitseva) [reply src/main/java/org/folio/qm/controller/RecordsEditorRecordsApi.java:33]: for the current story- yes

**Review by dmtkachenko — APPROVED** [review]

**Comment by dcrossleyau** [comment]
@natali-zaitseva @psmagin Regarding the main problem being reported by api-lint (which is also being reported in mainline branch):

I am no expert, but i tried this locally and folio-tools/api-lint tool is happy.
```
   examples:
     quickMarc:
-      $ref: examples/quickMarcJson.sample
+      value:
+        $ref: examples/quickMarcJson.sample
     error:
-      $ref: examples/error.sample
+      value:
+        $ref: examples/error.sample
```

**Comment by natali-zaitseva** [comment]
> @natali-zaitseva @psmagin Regarding the main problem being reported by api-lint (which is also being reported in mainline branch):
> 
> I am no expert, but i tried this locally and folio-tools/api-lint tool is happy.
> 
> ```
>    examples:
>      quickMarc:
> -      $ref: examples/quickMarcJson.sample
> +      value:
> +        $ref: examples/quickMarcJson.sample
>      error:
> -      $ref: examples/error.sample
> +      value:
> +        $ref: examples/error.sample
> ```

@dcrossleyau thank you!

**Review by psmagin — APPROVED** [review]

**Pull request merged**